tests/Cache/FileCacheManagerTest.php

```diff
@@ -351,38 +217,170 @@ public function testSetFileUsesRelativePathToFile(): void
-        $handlerProphecy->read()->willReturn($cache);
-        $handlerProphecy->getFile()->willReturn($cacheFile);
-        $handlerProphecy->write(Argument::is($cache));
-        $handler = $handlerProphecy->reveal();
+        $cache = $this->getCacheDouble($cachedSignature);
+        $handler = $this->getFileHandlerDouble($cache, $cacheFile);
 
-        $manager = new FileCacheManager(
-            $handler,
-            $signature,
-            false,
-            $directoryProphecy->reveal()
-        );
+        $manager = new FileCacheManager($handler, $signature, false, $directory);
 
         $manager->setFile($file, $fileContent);
+
+        self::assertTrue($cache->has($relativePathToFile));
+        self::assertSame(md5($fileContent), $cache->get($relativePathToFile));
     }
 
     private function getFile(): string
     {
         return __DIR__.'/../Fixtures/.php_cs.empty-cache';
     }
+
+    private function getDirectoryDouble(string $relativePathToFile): DirectoryInterface
+    {
+        return new class($relativePathToFile) implements DirectoryInterface {
+            private string $relativePathToFile;
+
+            public function __construct(string $relativePathToFile)
+            {
+                $this->relativePathToFile = $relativePathToFile;
+            }
+
+            public function getRelativePathTo(string $file): string
+            {
+                return $this->relativePathToFile;
+            }
+        };
+    }
+
+    private function getSignatureDouble(bool $isEqual): SignatureInterface
+    {
+        return new class($isEqual) implements SignatureInterface {
+            private bool $isEqual;
+
+            public function __construct(bool $isEqual)
+            {
+                $this->isEqual = $isEqual;
+            }
+
+            public function getPhpVersion(): string
+            {
+                throw new \LogicException('Not implemented.');
+            }
+
+            public function getFixerVersion(): string
+            {
+                throw new \LogicException('Not implemented.');
+            }
+
+            public function getIndent(): string
+            {
+                throw new \LogicException('Not implemented.');
+            }
+
+            public function getLineEnding(): string
+            {
+                throw new \LogicException('Not implemented.');
+            }
+
+            public function getRules(): array
+            {
+                throw new \LogicException('Not implemented.');
+            }
+
+            public function equals(SignatureInterface $signature): bool
+            {
+                return $this->isEqual;
+            }
+        };
+    }
+
+    /**
+     * @param array<string, string> $fileMap
+     */
+    private function getCacheDouble(SignatureInterface $signature, array $fileMap = []): CacheInterface
+    {
+        return new class($signature, $fileMap) implements CacheInterface {
```

Comment:
my habit speaking over me: this looks exactly like in-memory strategy implementation, ie `MemoryCache implements CacheInterface declared as normal class somewhere.
as long as it's single usage, OK with any way. if we would need it in multiple places, let's promote to non-anon class

(same for any double in this PR)

Reply:
looks like was not needed anywhere else